apps/nextjs/src/app/api/cron-jobs/expired-exports/route.ts

```diff
@@ -15,47 +15,53 @@ requiredEnvVars.forEach((envVar) => {
   }
 });
 
-async function updateExpiredAt(fileIds: string[]) {
+async function updateExpiredAtAndDelete(fileIds: string[]) {
   if (fileIds.length === 0) {
     log.info("No file IDs to update.");
     return;
   }
 
-  try {
-    const result = await prisma.lessonExport.updateMany({
-      where: {
-        gdriveFileId: {
-          in: fileIds,
-        },
-      },
-      data: {
-        expiredAt: new Date(),
-      },
-    });
-    log.info(`Updated expiredAt for ${fileIds.length} files.`);
+  const failedIds: string[] = [];
 
-    if (result.count === fileIds.length) {
-      log.info("All files updated successfully.");
-    } else {
-      throw new Error(
-        `Expected to update ${fileIds.length} files, but only updated ${result.count}.`,
-      );
-    }
-  } catch (error) {
-    log.error("Error updating expiredAt field in the database:", error);
-    throw error;
-  }
-}
+  for (const id of fileIds) {
+    try {
+      const record = await prisma.lessonExport.findFirst({
+        where: { gdriveFileId: id },
```

Comment:
Not unique value in database so find first to get lesson id

Reply:
Would we not want to get all rather than just the first one?

Reply:
> I could findAll and throw error if it finds more than one
Could we not findAll and updateMany?
Other option would be to add a unique constraint on that field

Reply:
The trouble with update many is that it only returns a count of the updated ids and not any info on failed ids

Reply:
Sure, fair enough. Let's put a unique constraint on it then, as we're not expecting multiple records per file! I'll put that in a separate PR